$NetBSD: patch-ab,v 1.9 2017/01/02 14:13:32 schmonz Exp $

Avoid "echo -n" in subshells and unsupported linker flags to fix OS X build.

--- Makefile.orig	2015-02-06 23:57:34.000000000 +0000
+++ Makefile
@@ -69,7 +69,7 @@ adt/hashb.lo adt/hashb.o: ltcompile adt/
 adt/hashs.lo adt/hashs.o: ltcompile adt/hashs.c include/bglibs/adt_common.h
 	./ltcompile adt/hashs.c
 
-all: sysdeps.h libraries programs man selftests
+all: sysdeps.h libraries programs man
 
 base64/asc2bin.lo base64/asc2bin.o: ltcompile base64/asc2bin.c include/bglibs/base64.h include/bglibs/str.h sysdeps.h
 	./ltcompile base64/asc2bin.c
@@ -90,7 +90,7 @@ base64/encode_part.lo base64/encode_part
 	./ltcompile base64/encode_part.c
 
 bg-installer: bg-installer.o ltload libbg-cli.la libbg.la bg-installer-cli.o
-	./ltload bg-installer libbg-cli.la libbg.la bg-installer-cli.o -lbg-cli -static 
+	./ltload bg-installer libbg-cli.la libbg.la bg-installer-cli.o -static
 
 bg-installer-cli.c: bg-installer.cli cli-generate sysdeps.h
 	./cli-generate -c $< >$@
@@ -157,7 +157,7 @@ compile: conf-cc
 	chmod 755 compile
 
 crc-gentab: crc-gentab.o load fmt/lcase.o fmt/ucase.o fmt/unum.o iobuf/iobuf_timeout.o iobuf/obuf_flush.o iobuf/obuf_pad.o iobuf/obuf_put3s.o iobuf/obuf_put5s.o iobuf/obuf_putc.o iobuf/obuf_putunumw.o iobuf/obuf_stderr.o iobuf/obuf_stdout.o iobuf/obuf_write.o msg/common.o msg/die.o sys/iopoll.o
-	./load crc-gentab fmt/lcase.o fmt/ucase.o fmt/unum.o iobuf/iobuf_timeout.o iobuf/obuf_flush.o iobuf/obuf_pad.o iobuf/obuf_put3s.o iobuf/obuf_put5s.o iobuf/obuf_putc.o iobuf/obuf_putunumw.o iobuf/obuf_stderr.o iobuf/obuf_stdout.o iobuf/obuf_write.o msg/common.o msg/die.o sys/iopoll.o -static 
+	./load crc-gentab fmt/lcase.o fmt/ucase.o fmt/unum.o iobuf/iobuf_timeout.o iobuf/obuf_flush.o iobuf/obuf_pad.o iobuf/obuf_put3s.o iobuf/obuf_put5s.o iobuf/obuf_putc.o iobuf/obuf_putunumw.o iobuf/obuf_stderr.o iobuf/obuf_stdout.o iobuf/obuf_write.o msg/common.o msg/die.o sys/iopoll.o
 
 crc-gentab.o: compile crc-gentab.c sysdeps.h include/bglibs/iobuf.h include/bglibs/iobuf_common.h include/bglibs/ibuf.h include/iobuf_common.h include/bglibs/obuf.h sysdeps.h include/bglibs/msg.h sysdeps.h include/bglibs/debug.h
 	./compile crc-gentab.c
@@ -293,7 +293,7 @@ dl.lib: compile load
 	@echo 'main() { ; }' >trylib-ldl.c
 	@{ ./compile trylib-ldl.c && ./load trylib-ldl -ldl; } >/dev/null 2>&1 \
 	  && { echo -ldl >dl.lib; echo yes; } \
-	  || { echo -n >dl.lib; echo no; }
+	  || { : >dl.lib; echo no; }
 	@rm -f trylib-ldl.c trylib-ldl.o trylib-ldl
 
 fmt/char.lo fmt/char.o: ltcompile fmt/char.c include/bglibs/fmt.h
@@ -863,9 +863,7 @@ path/mktemp.lo path/mktemp.o: ltcompile 
 	./ltcompile path/mktemp.c
 
 perl-head.pl: 
-	( set -e; PATH="/bin:/usr/bin:/usr/local/bin:$$PATH"; export PATH; \
-	  perl=`which perl`; \
-	  echo "#! $$perl"; \
+	( echo "#! ${PERL5}"; \
 	  echo "# WARNING: This file was auto-generated. Do not edit!"; \
 	  echo ) >perl-head.pl
 
@@ -885,7 +883,7 @@ rt.lib: compile load
 	@echo 'main() { ; }' >trylib-lrt.c
 	@{ ./compile trylib-lrt.c && ./load trylib-lrt -lrt; } >/dev/null 2>&1 \
 	  && { echo -lrt >rt.lib; echo yes; } \
-	  || { echo -n >rt.lib; echo no; }
+	  || { : >rt.lib; echo no; }
 	@rm -f trylib-lrt.c trylib-lrt.o trylib-lrt
 
 selftest-cmp: selftest-cmp.o ltload libbg.la
